### ISSUE TITLE
latest commit breaks cmake

### ISSUE THREAD

**Issue by mjsduncan** [opencog/opencog#1928]
hi guys!
````
cmake ..
````
...
````
The following components will be built:
-----------------------------------------------
   Attention        - Agents for attention allocation dynamics.
   Blopex           - BLOPEX block eigenvalue solver.
   Cython bindings  - Cython (python) bindings.
   Dim. Embedding   - Compute dimensional embedding of the AtomSpace.
   Doxygen          - Code documentation.
   Python tests     - Python bindings nose tests.
   REST             - REST Atomspace Publisher module.
   Server           - The CogServer.
   Unit tests       - Unit tests.

The following components WILL NOT be built:
-----------------------------------------------
   GTK Visualizer   - GTK3-based Atomspace Visualizer.
   Spatial          - Vizualisation of the 3D spacemap.
   Viterbi          - Viterbi Parser for Link Grammar.

CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
ATOMSPACE_truthvalue_LIBRARY
    linked by target "attention" in directory /home/biocog/opencog/opencog/dynamics/attention
    linked by target "PersistModule" in directory /home/biocog/opencog/opencog/modules
    linked by target "PersistZmqModule" in directory /home/biocog/opencog/opencog/modules
    linked by target "QueryModule" in directory /home/biocog/opencog/opencog/modules
    linked by target "RuleEngineModule" in directory /home/biocog/opencog/opencog/modules
    linked by target "PythonModule" in directory /home/biocog/opencog/opencog/modules/python
    linked by target "server" in directory /home/biocog/opencog/opencog/server
    linked by target "agent_finder" in directory /home/biocog/opencog/opencog/cython/opencog
    linked by target "cogserver_cython" in directory /home/biocog/opencog/opencog/cython/opencog
    linked by target "cogserver_type_constructors" in directory /home/biocog/opencog/opencog/cython/opencog
    linked by target "AtomSpacePublisherModuleUTest" in directory /home/biocog/opencog/tests/persist/zmq/events
    linked by target "PLNRulesUTest" in directory /home/biocog/opencog/tests/reasoning
    linked by target "PyEvalUTest" in directory /home/biocog/opencog/tests/cython
    linked by target "PythonModuleUTest" in directory /home/biocog/opencog/tests/cython
    linked by target "hopfield" in directory /home/biocog/opencog/examples/hopfield

-- Configuring incomplete, errors occurred!
````
i did a fresh clone to make sure i hadn't mucked something up myself.

**Comment by linas** [comment]
ah, you have to pull, build and install a new atomspace first!   Sorry! 

I wonder if there was some transitional way to handle this without forcing everyone to do this...

**Comment by linas** [comment]
Ah .. @ngeiswei  there is a  `FIND_PACKAGE(AtomSpace 5.0.3 REQUIRED)` in the opencog master CMakefile, and maybe that should have been changed to 5.1.0 to make it clear that the dependency changed.  a code restructure like this certainly does merit a minor-version-number bump (technically, a major-version bump, according to autotools, but whatever).

**Comment by linas** [comment]
Hmm Just looked at lib/FindAtomSpace.cmake -- it does not verify version numbers.

Just looked at opencog/atomspace/version.h -- no one seems to actually do anything with those version numbers.

Just looked at `/usr/local/lib/opencog` and NONE of those libraries are versioned!   

So I guess CMake has versioning turned off by default.  It wouldn't really solve this problem, but it would at least make it less ugly.  Given that I wrote the above two, I guess I ran out of steam before I figured out the magic CMake incantation to make it turn on versioning.
